client/components/MainSettings.jsx

```diff
@@ -19,45 +20,62 @@ const MainSettings = React.createClass({
     });
   },
 
-  componentWillUnmount() {
-    Meteor.call("room.archive", this.props.room._id, this.state.archived);
-  },
-
   toggleArchivedState() {
-    Meteor.call("room.archive", this.props.room._id, !this.state.archived);
+    const params = {
+      archived: !this.state.archived,
+      roomId: this.props.room._id
+    };
+    Meteor.call("room.archive", params);
     this.setState({archived: !this.state.archived});
   },
 
   render() {
-    const roomUsersHTML = this.state.roomUsers.map(function(user) {
+    const user = Meteor.user();
```

Comment:
moved it to the meteor data, noted importance of not calling reactive methods in render